opentelemetry-sdk/src/metrics/internal/aggregate.rs

```diff
@@ -83,28 +94,24 @@ impl<T: Number> AggregateBuilder<T> {
         }
     }
 
-    /// Wraps the passed in measure with an attribute filtering function.
-    fn filter(&self, f: impl Measure<T>) -> impl Measure<T> {
-        let filter = self.filter.clone();
-        move |n, attrs: &[KeyValue]| {
-            if let Some(filter) = &filter {
-                let filtered_attrs: Vec<KeyValue> =
-                    attrs.iter().filter(|kv| filter(kv)).cloned().collect();
-                f.call(n, &filtered_attrs);
-            } else {
-                f.call(n, attrs);
-            };
+    fn maybe_filtered_measurement(&self, instrument: Arc<impl Measure<T>>) -> Arc<dyn Measure<T>> {
```

Comment:
Any advice on naming?
both for `maybe_filtered_measurement` function, and for `FilteredMeasureInstrument`...